facet-reflect/src/wip/mod.rs

```diff
@@ -283,15 +283,12 @@ pub struct Wip<'a> {
     istates: FlatMap<ValueId, IState>,
 
     /// lifetime of the shortest reference we hold
-    phantom: PhantomData<&'a ()>,
+    phantom: PhantomData<*mut &'a ()>,
```

Comment:
nit but I would define this as follows (and do this elsewhere as well)
```suggestion
    phantom: PhantomData<fn(&'a ()) -> &'a ()>,
```
Expressing variance with function pointers it a bit more generic, `fn(a)` = contravariant, `fn() -> a` covariant, `fn(a) -> a` invariant